### PR TITLE
[NPU] Support mlaprolog, fp8 KVcache(only MLA) and ChunkPrefill for A5(950PR/DT) NPU

### PR DESCRIPTION
**Pull request by LinyuanLi0046**
## Motivation

This PR adds Ascend NPU support for MLAProlog with FP8 KV cache on Atlas A5 hardware, and enables the required A5 chunk prefill path for MLA attention. The goal is to support ModelSlim `Q_FP8_DYNAMIC_KV_FP8` checkpoints and run DeepSeek-style MLA workloads with FP8 KV cache correctly on A5, while keeping the migration limited to the necessary NPU attention, KV cache, graph metadata, and weight preprocessing paths.

## Modifications

- Added Atlas A5 hardware detection utility for NPU runtime path selection.
- Added ModelSlim KV quantization scheme support for `Q_FP8_DYNAMIC_KV_FP8`, including FA Q/K/V scale parameter registration and post-load runtime scale preparation.
- Preserved original MXFP8 linear weight and scale tensors after loading so MLAProlog preprocessing can consume the original layout/scale metadata.
- Updated DeepSeek MLA initialization and weight loading to create KV quantization parameters and refresh FA-K scale runtime parameters after `kv_b_proj` processing.
- Extended NPU MLA preprocessing to support MLAProlog FP8 KV cache:
  - handles MXFP8 query quantization;
  - prepares original quantized weights and scales for MLAProlog;
  - passes FP8 KV cache quant/dequant scales into `npu_mla_prolog_v3`;
  - standardizes MLA preprocess return values for downstream attention paths.
- Updated NPU DeepSeek MLA attention flow to propagate FP8 KV runtime scales and `dequant_scale_q_nope` into the Ascend attention backend.
- Updated Ascend attention backend for FP8 KV cache MLA decode/graph/MTP paths using `npu_fused_infer_attention_score_v2`.
- Added A5-specific MLA chunk prefill prefix-cache path that materializes prefix KV cache and applies FP8 dequantization before `kv_b_proj`.
- Updated NPU MLA KV memory pool layout so FP8 KV cache stores C-KV and K-Rope with the required dtype handling.
- Updated NPU graph runner v2 selection for MLA FP8 KV cache graph updates.

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: **Missing `run-ci` label** -- add it to run CI tests.
Latest PR Test (Extra): :x: **Blocked** -- `run-ci` is required first.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 1]
```diff
@@ -32,13 +34,30 @@
...
+    def create_weights(self, layer: torch.nn.Module, *args, **kwargs):
+        self.scheme.create_weights(layer, *args, **kwargs)
+
+    def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
+        self.scheme.process_weights_after_loading(layer)
+
+    def apply(self, layer: torch.nn.Module, *args, **kwargs) -> torch.Tensor:
+        raise RuntimeError(
+            "ModelSlimQFP8DynamicKVFP8Method.apply should not be called."
+        )
```
Comment (TamirBaydasov) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py:58]: The Quant Method abstraction should be more generic, i.e. `ModelSlimKVMethod`. And `scheme` should be a property of a layer.

I also suggest to look into implementing `apply` method here so we call it later in memory pooler and quantize kv cache during `set_kv_buffer`. Will that work for DeepSeek?

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 2]
```diff
@@ -155,6 +174,10 @@ def get_quant_method(
         from sglang.srt.layers.linear import LinearBase
         from sglang.srt.layers.moe.fused_moe_triton import FusedMoE
 
+        kv_method = self._maybe_get_kv_method(layer, prefix)
+        if kv_method is not None:
+            return kv_method
```
Comment (TamirBaydasov) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py:179]: This functionality should be applied only for the layers we are interested in. In this case it should be `RadixAttention`, I believe. You can follow examples from compressed-tensors and quark classes.

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 3]
```diff
@@ -185,6 +208,29 @@ def get_quant_method(
...
+    def _maybe_get_kv_method(
+        self, layer: torch.nn.Module, prefix: str
+    ) -> Optional[QuantizeMethodBase]:
+        kv_quant_type = self.quant_description.get(f"{prefix}.quant_type")
+        if kv_quant_type != "Q_FP8_DYNAMIC_KV_FP8":
+            return None
+
+        return ModelSlimQFP8DynamicKVFP8Method(
+            self.get_kv_scheme(layer, prefix, kv_quant_type)
+        )
```
Comment (TamirBaydasov) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py:232]: This method should be called with config class as input parameter, while scheme should be applied to layer itself, like it is done for other layer types.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py [hunk 4]
```diff
@@ -280,6 +291,7 @@ def __init__(self, model_runner: ModelRunner, speculative_step_id: int = 0):
         super().__init__()
         self.forward_metadata = None
         self.device = model_runner.device
+        self.fp8_kv_scale = torch.ones(1, dtype=torch.float32, device=self.device)
```
Comment (TamirBaydasov) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py:294]: Why do we need this value here? Scales should come from layer and be handled inside scheme.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py [hunk 5]
```diff
@@ -1088,6 +1277,8 @@ def forward_extend(
         topk_indices: Optional[torch.Tensor] = None,
         sinks: Optional[torch.Tensor] = None,
         slopes: Optional[torch.Tensor] = None,
+        dequant_scale_q_nope: Optional[torch.Tensor] = None,
+        fp8_kv_scale: Optional[torch.Tensor] = None,
```
Comment (TamirBaydasov) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py:1281]: Ditto https://github.com/sgl-project/sglang/pull/29641/changes#r3684455133
Would it be possible to store `fp8_kv_scale `inside `layer.scheme` here?

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py [hunk 6]
```diff
@@ -2043,6 +2257,59 @@ def forward_mtp(
...
+                    sparse_mode=3,
+                    atten_mask=self.mtp_mask,
+                    dequant_scale_query=dequant_scale_query,
+                    dequant_scale_key=kv_scale,
+                    dequant_scale_value=kv_scale,
+                    key_quant_mode=0,
+                    value_quant_mode=0,
+                    query_quant_mode=3,
+                    out=[attn_output, softmax_lse],
+                )
```
Comment (TamirBaydasov) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py:2293]: Do I understand correctly that query type is `fp8` , so the operator is run in full quant scenario?

**Review by TamirBaydasov — CHANGES_REQUESTED** [review]
Hi! I looked through the changes and collected some questions. Could you look into them?

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py (hunk 1 above)
Comment (LinyuanLi0046) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:58]: The NPU memory pool is currently being refactored by the community. Once completed, this PR will be redesigned based on the new code.
